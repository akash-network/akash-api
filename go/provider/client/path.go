package rest

import (
	"fmt"

	mtypes "github.com/akash-network/akash-api/go/node/market/v1beta4"
)

const (
	DeploymentPathPrefix = "/deployment/{dseq}"
	LeasePathPrefix      = "/lease/{dseq}/{gseq}/{oseq}"
	HostnamePrefix       = "/hostname"
	EndpointPrefix       = "/endpoint"
	MigratePathPrefix    = "/migrate"
)

func VersionPath() string {
	return "version"
}

func StatusPath() string {
	return "status"
}

func ValidatePath() string {
	return "validate"
}

func LeasePath(id mtypes.LeaseID) string {
	return fmt.Sprintf("lease/%d/%d/%d", id.DSeq, id.GSeq, id.OSeq)
}

func SubmitManifestPath(dseq uint64) string {
	return fmt.Sprintf("deployment/%d/manifest", dseq)
}

func GetManifestPath(id mtypes.LeaseID) string {
	return fmt.Sprintf("%s/manifest", LeasePath(id))
}

func LeaseStatusPath(id mtypes.LeaseID) string {
	return fmt.Sprintf("%s/status", LeasePath(id))
}

func LeaseShellPath(lID mtypes.LeaseID) string {
	return fmt.Sprintf("%s/shell", LeasePath(lID))
}

func LeaseEventsPath(id mtypes.LeaseID) string {
	return fmt.Sprintf("%s/kubeevents", LeasePath(id))
}

func ServiceStatusPath(id mtypes.LeaseID, service string) string {
	return fmt.Sprintf("%s/service/%s/status", LeasePath(id), service)
}

func ServiceLogsPath(id mtypes.LeaseID) string {
	return fmt.Sprintf("%s/logs", LeasePath(id))
}
