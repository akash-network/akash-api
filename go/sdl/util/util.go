package util

import (
	"math"

	sdk "github.com/cosmos/cosmos-sdk/types"

	rtypes "pkg.akt.dev/go/node/types/resources/v1beta4"
)

func ComputeCommittedResources(factor float64, rv rtypes.ResourceValue) rtypes.ResourceValue {
	// If the value is less than 1, commit the original value. There is no concept of under-commit
	if factor <= 1.0 {
		return rv
	}

	v := rv.Val.Uint64()
	fraction := 1.0 / factor
	committedValue := math.Round(float64(v) * fraction)

	// Don't return a value of zero, since this is used as a resource request
	if committedValue <= 0 {
		committedValue = 1
	}

	result := rtypes.ResourceValue{
		Val: sdk.NewInt(int64(committedValue)),
	}

	return result
}
