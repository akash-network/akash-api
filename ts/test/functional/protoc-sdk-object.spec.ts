import { describe, expect, it } from "@jest/globals";
import { exec } from "child_process";
import { readFile, rmdir } from "fs/promises";
import { tmpdir } from "os";
import { join as joinPath } from "path";
import { promisify } from "util";

const execAsync = promisify(exec);

describe("protoc-sdk-objec plugin", () => {
  const config = {
    version: "v2",
    clean: true,
    plugins: [
      {
        local: "bin/protoc-sdk-object.ts",
        strategy: "all",
        out: ".",
        opt: [
          "target=ts",
        ],
      },
    ],
  };

  it("generates SDK object from proto files", async () => {
    const outputDir = joinPath(tmpdir(), `ts-bufplugin-${process.pid.toString()}`);
    const command = `buf generate --template '${JSON.stringify(config)}' -o '${outputDir}' ./test/functional/proto`;

    try {
      await execAsync(command, {
        cwd: joinPath(__dirname, "..", ".."),
        env: {
          ...process.env,
          BUF_PLUGIN_SDK_OBJECT_OUTPUT_FILE: "sdk.ts",
        },
      });

      expect(await readFile(joinPath(outputDir, "sdk.ts"), "utf-8")).toMatchSnapshot();
    } finally {
      await rmdir(outputDir, { recursive: true });
    }
  });
});
