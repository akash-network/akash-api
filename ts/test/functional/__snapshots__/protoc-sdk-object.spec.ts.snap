// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`protoc-sdk-objec plugin generates SDK object from proto files 1`] = `
"import type * as msg_pb from './protos/./msg_pb'
import type * as query_pb from './protos/./query_pb'
import type { ClientFactory } from '../sdk/ClientFactory';
import type { CallOptions, TxCallOptions } from '../transport';
import { createServiceLoader } from '../utils/createServiceLoader';
import { withMetadata } from '../utils/sdkMetadata';


export const serviceLoader = createServiceLoader([
  () => import('./protos/msg_pb').then(m => m.Msg),
  () => import('./protos/query_pb').then(m => m.Query)
] as const);
export function createSDK<T extends ClientFactory>(clientFactory: T) {
  return {
    test: {
      functional: {
        proto: {
          send: withMetadata(async function send(input: msg_pb.SendRequestJson, options?: TxCallOptions) {
            const service = await serviceLoader.loadAt(0);
            return clientFactory.getClient(service).send(input, options);
          }, { path: [0, 0] }),
          getBlock: withMetadata(async function getBlock(input: query_pb.GetBlockRequestJson, options?: CallOptions) {
            const service = await serviceLoader.loadAt(1);
            return clientFactory.getClient(service).getBlock(input, options);
          }, { path: [1, 0] }),
          /**
           * getBlockByHeight is deprecated. Use GetBlock instead.
           * @deprecated
           */
          getBlockByHeight: withMetadata(async function getBlockByHeight(input: query_pb.GetBlockByHeightRequestJson, options?: CallOptions) {
            const service = await serviceLoader.loadAt(1);
            return clientFactory.getClient(service).getBlockByHeight(input, options);
          }, { path: [1, 1] })
        }
      }
    }
  }
}
"
`;
